packages/template-retail-react-app/app/components/search/index.jsx

```diff
@@ -151,6 +163,33 @@ const Search = (props) => {
         setIsOpen(false)
     }
 
+    const launchChat = () => {
+        window.embeddedservice_bootstrap.utilAPI
+            .launchChat()
+            .then((successMessage) => {
+                /* TODO: With the Salesforce Winter '26 release, we will be able to use the
+                 * onEmbeddedMessagingFirstBotMessageSent event instead, and get rid of this logic. */
+                if (successMessage.includes('Successfully initialized the messaging client')) {
+                    hasFired = false //We want the logic in onEmbeddedMessageSent to happen once per new conversation
+                    newChatLaunched = true
+                }
+            })
+            .catch((err) => {
+                console.error('launchChat error', err)
+            })
+
+        window.addEventListener('onEmbeddedMessageSent', (e) => {
```

Comment:
Every time `launchChat` is called, a new event listener is attached. If not necessary, let's make sure to attach the listener once.